packages/components/src/search-control/style.scss

```diff
@@ -41,8 +41,15 @@
 		height: $grid-unit-50;
 	}
 
-	&.is-size-compact input[type="search"].components-search-control__input {
-		height: $grid-unit-40;
+	&.is-size-compact {
```

Comment:
Nit: let's reuse the `components-search-control__icon` block a few lines below and rewrite styles that way:

```diff
diff --git a/packages/components/src/search-control/style.scss b/packages/components/src/search-control/style.scss
index 632fc851c9..d70fd2dc68 100644
--- a/packages/components/src/search-control/style.scss
+++ b/packages/components/src/search-control/style.scss
@@ -41,17 +41,10 @@
 		height: $grid-unit-50;
 	}
 
-	&.is-size-compact {
-		input[type="search"].components-search-control__input {
-			height: $grid-unit-40;
-			padding-left: $grid-unit-10;
-		}
-
-		.components-search-control__icon {
-			right: $grid-unit-05;
-		}
+	&.is-size-compact input[type="search"].components-search-control__input {
+		height: $grid-unit-40;
+		padding-left: $grid-unit-10;
 	}
-
 }
 
 .components-search-control__icon {
@@ -65,6 +58,10 @@
 	justify-content: center;
 
 	width: $icon-size;
+
+	.is-size-compact & {
+		right: $grid-unit-05;
+	}
 }
 
 .components-search-control__input-wrapper {

```